packages/appium-tizen-tv-driver/lib/rc-pair.js

```diff
@@ -2,45 +2,70 @@ import yargs from 'yargs/yargs';
 import {TizenRemote, Keys as KEYS} from '@headspinio/tizen-remote';
 import {RC_OPTS} from './driver';
 
-export async function pairRemote({host}) {
-  const rc = new TizenRemote(host, RC_OPTS);
+/**
+ *
+ * @param {{host:string, port?: number}} opts
+ * @returns {Promise<void>}
+ */
+export async function pairRemote({host, port}) {
+  const rc = new TizenRemote(host, {...RC_OPTS, port});
 
-  const token = await rc.getToken();
-  if (token) {
-    console.log(token); // eslint-disable-line no-console
-    return;
+  try {
+    const token = await rc.getToken({force: true});
+    if (token) {
+      console.log(token); // eslint-disable-line no-console
+      return;
+    }
+    throw new Error(`Could not retrieve token; please try allowing the remote again`);
+  } finally {
+    await rc.disconnect();
   }
-
-  throw new Error(`Could not retrieve token; please try allowing the remote again`);
 }
 
-export async function testRemote({host, token}) {
+/**
+ *
+ * @param {{host: string, token:string, port?: number}} opts
+ */
+export async function testRemote({host, token, port}) {
   const rc = new TizenRemote(host, {
     ...RC_OPTS,
+    port,
     token,
   });
+  try {
+    await rc.click(KEYS.HOME);
+    console.log('OK'); // eslint-disable-line no-console
+  } finally {
+    await rc.disconnect();
+  }
 
-  await rc.click(KEYS.HOME);
-  await rc.disconnect();
 }
 
 async function main() {
-  const {host, token} = yargs(process.argv.slice(2))
+  const {host, token, port} = await yargs(process.argv.slice(2))
```

Comment:
wow since when is yargs parse async??

Reply:
I don't think it actually is, but it seems to help typescript understand the return value